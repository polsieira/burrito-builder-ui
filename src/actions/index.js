export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const addOrders = newOrder => ({
  type: 'ADD_ORDERS',
  newOrder
});
