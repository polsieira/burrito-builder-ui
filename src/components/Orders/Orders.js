import React, { Component } from 'react';
import './Orders.css';
import { getOrders } from '../../apiCalls';
import Order from '../Order/Order';
import { connect } from 'react-redux';
import { setOrders } from '../../actions';
import { bindActionCreators } from 'redux';


export class Orders extends Component {

  componentDidMount() {
    getOrders()
      .then(data => this.props.setOrders(data.orders))
      .catch(err => console.error('Error fetching:', err));
  }

  render() {
    const { orders } = this.props;
    const orderEls = orders.map(order => {
      return (
        <Order order={order} key={order.id} />
      )
    });
    return (
      <section>
        {orderEls.length ? orderEls : <p>No orders yet!</p>}
      </section >
    )
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    setOrders
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Orders);