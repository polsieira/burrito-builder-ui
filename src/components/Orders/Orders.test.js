import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders';
import { setOrders } from '../../actions';

// jest.mock('../../apiCalls.js');

// Sorry about the nested describe blocks, I like them

describe('Orders', () => {

  describe('Orders container', () => {
    let wrapper;
    const mockOrders = [
      {
        id: 1,
        name: 'Pol',
        ingredients: ['beans', 'more beans', 'more more beans']
      },
      {
        id: 2,
        name: 'Lucy',
        ingredients: ['beans', 'more beans', 'more more beans']
      },
      {
        id: 3,
        name: 'Kate',
        ingredients: ['beans', 'more beans', 'more more beans']

      }
    ];
    const mockSetOrders = jest.fn();

    beforeEach(() => {
      window.fetch = jest.fn().mockImplementation(() => {
        return Promise.resolve({
          ok: true,
          json: () => Promise.resolve(mockOrders)
        });
      });
      wrapper = shallow(<Orders orders={mockOrders} setOrders={mockSetOrders} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });


  describe('mapStateToProps', () => {
    it('should return an object with orders', () => {
      const mockOrders = [
        {
          id: 1,
          name: 'Pol',
          ingredients: ['beans', 'more beans', 'more more beans']
        },
        {
          id: 2,
          name: 'Lucy',
          ingredients: ['beans', 'more beans', 'more more beans']
        },
        {
          id: 3,
          name: 'Kate',
          ingredients: ['beans', 'more beans', 'more more beans']

        }
      ]
      const mockState = {
        orders: mockOrders,
        errorMsg: 'You got an error fool',
        isLoading: false
      };

      const expected = {
        orders: mockOrders
      }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    it('calls dispatch with an setOrders action when componentDidMount is called', () => {
      const mockOrders = [
        {
          id: 1,
          name: 'Pol',
          ingredients: ['beans', 'more beans', 'more more beans']
        },
        {
          id: 2,
          name: 'Lucy',
          ingredients: ['beans', 'more beans', 'more more beans']
        },
        {
          id: 3,
          name: 'Kate',
          ingredients: ['beans', 'more beans', 'more more beans']

        }
      ];
      const mockDispatch = jest.fn();
      const actionToDispatch = setOrders(mockOrders);

      // Execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setOrders(mockOrders);

      // Expectaion
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});