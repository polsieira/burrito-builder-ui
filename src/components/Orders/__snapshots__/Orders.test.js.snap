// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Orders container should match the snapshot 1`] = `
<section>
  <Order
    handleDeleteOrder={[Function]}
    key="1"
    order={
      Object {
        "id": 1,
        "ingredients": Array [
          "beans",
          "more beans",
          "more more beans",
        ],
        "name": "Pol",
      }
    }
  />
  <Order
    handleDeleteOrder={[Function]}
    key="2"
    order={
      Object {
        "id": 2,
        "ingredients": Array [
          "beans",
          "more beans",
          "more more beans",
        ],
        "name": "Lucy",
      }
    }
  />
  <Order
    handleDeleteOrder={[Function]}
    key="3"
    order={
      Object {
        "id": 3,
        "ingredients": Array [
          "beans",
          "more beans",
          "more more beans",
        ],
        "name": "Kate",
      }
    }
  />
</section>
`;
